source/get-response.ts

```diff
@@ -15,7 +15,11 @@ export default (response: IncomingMessage, options: Options, emitter: EventEmitt
 
 	mimicResponse(response, progressStream);
 
-	const newResponse: PassThrough | Transform = options.decompress === true &&
+	// @todo
+	// I believe this typings was slightly wrong because the way `response` is used in 
+	// `as-stream.ts::52` it requires a `statusCode` which `IncomingMessage` from node has.
+	// Maybe a maintainer can guide me a bit here.
+	const newResponse: Response = options.decompress === true &&
```

Comment:
I'd like to confirm this change makes sense. I'm not super knowledgeable about what the actual code is doing. If someone could clarify this for me.

Reply:
Correct. Good catch.